docs/audio/Soundfont_XML.md

```diff
@@ -0,0 +1,314 @@
+
+            **Properties**
+
+            - **Delay**: Duration until the next point
+            - **Arg**: Value of the envelope at this point
+
+            ---
+
+        -
+            ```xml
+            <Disable/>
+            ```
+            Insert a ADSR_DISABLE command
+
+            ---
+
+        -
+            ```xml
+            <Hang/>
+            ```
+            Insert a ADSR_HANG command
+
+            ---
+
+        -
+            ```xml
+            <Goto
+                Index="<uint>"
+            />
+            ```
+            Insert a ADSR_GOTO command
+
+            **Properties**
+
+            - **Index**: Index of the envelope point to jump to
+
+            ---
+
+        ```xml
+        </Envelope>
+        ```
+        ---
+
+    ```xml
+    </Envelopes>
+    ```
+    ---
+
+-
+    ```xml
+    <Samples
+        IsDD="[Bool]"
+        Cached="[Bool]"
+    >
+    ```
+    Begins a list of samples used in this Soundfont.
+
+    **Properties**
+
+    - <ins>[Optional]</ins> **IsDD**: Whether all the samples in the list are on the Disk Drive. The sample data will come from the samplebank `SampleBankDD`. **Default is `false`.** **NOTE this is not fully implemented, it should always be `false`.**
+    - <ins>[Optional]</ins> **Cached**: Whether all the samples in the list should be added to the `usedSamples` cache. **Default is `false`.**
+
+    **Sub-Attributes**
+
+    -
+        ```xml
+        <Sample
+            Name="<C Identifier>"
+            SampleRate="[Sample Rate]"
+            BaseNote="[Note Number]"
+            IsDD="[Bool]"
+            Cached="[Bool]"
+        />
+        ```
+        Declares a sample used in this soundfont.
+
+        **Properties**
+
+        - **Name**: The name of this sample. A sample with this name must be present in the samplebank used by the soundfont.
+        - <ins>[Optional]</ins> **SampleRate**: An overriding sample rate for this sample. **Default comes from the sample file.**
+        - <ins>[Optional]</ins> **BaseNote**: An overriding root key for this sample. **Default comes from the sample file.**
+        - <ins>[Optional]</ins> **IsDD**: Whether this sample is on the Disk Drive. The sample data will come from the samplebank `SampleBankDD`. **Default is `false`.** **NOTE this is not fully implemented, it should always be `false`.**
+        - <ins>[Optional]</ins> **Cached**: Whether this sample should be added to the `usedSamples` cache. **Default is `false`.**
+
+        ---
+
+    ```xml
+    </Samples>
+    ```
+    ---
+
+-
+    ```xml
+    <Effects>
+    ```
+    Begins a list of sound effects to define for this soundfont. Sound effects correspond to simple sounds that cannot be played at different keys.
+
+    **Properties**
+
+    N/A
+
+    **Sub-Attributes**
+
+    -
+        ```xml
+        <Effect
+            Name="<C Identifier>"
+            Sample="<Sample Name>"
+            SampleRate="[Sample Rate]"
+            BaseNote="[Note Number]"
```

Comment:
Define "Note Number"? If I didn't know better I'd attempt to put "C4" or something here

Reply:
C4 would actually be correct, I've changed it to say `Note Name` and added a brief explanation of the field at the top of the file.